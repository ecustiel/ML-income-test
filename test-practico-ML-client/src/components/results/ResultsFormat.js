import React from 'react'
import { useHistory } from "react-router-dom";

export const ResultsFormat = (items) => {
    
    const history = useHistory();

    const handleClick = (e) => {
        e.preventDefault();
        const id = items.items[0].id;
        history.push(`/items/${id}`);
    }
    

    return (
        <>
        <div className='resultsFormat__container pointer' onClick={handleClick} >
            <div className='resultsFormat__imgContainer'>
                <img src={items.items[0].picture} alt='prodImg' className='resultsFormat__imgContainer-img' />
            </div>
            <div className='resultsFormat'>
            <div className='resultsFormat__detsContainer'>
                <p className='resultsFormat_detsContainer-price'>$. {items.items[0].price.amount}</p>
                {
                    (items.items[0].free_shipping) &&
                        <img src='./Assets/ic_shipping.png' alt='shipp' className='resultsFormat_detsContainer-img' />    
                }
            </div>
            
            <div className='resultsFormat__titleContainer'>
                <p className='resultsFormat__titleContainer-title'>{items.items[0].title}</p>
            </div>
            </div>
            <div className='resultsFormat__locContainer'>
                <p className='resultsFormat__locContainer-loc'>{items.items[0].loc}</p>
            </div>
            
            
        </div>
        </>
    )
}
