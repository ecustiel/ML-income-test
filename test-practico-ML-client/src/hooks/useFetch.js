import {useState, useEffect, useRef} from 'react';


export const useFetch = (url) => {

   
    const isMounted = useRef(true);
    const [state, setState] = useState({data: null, loading: true, error: null});

    
    useEffect(() => {
        
        return () => {
            isMounted.current = false;    
        }
    }, [])

    useEffect(() => {

        setState({data: null, loading: true, error: null});
        
        fetch( url )
        .then( resp => resp.json())
        .then( data => {

            if(isMounted.current) {
            setState({
                loading: false,
                error: null,
                data
            })
        }

        })
        .catch(() => {
            setState({
                loading: false,
                error: 'No se pudo cargar la informacion',
                data: null
            })
        })
        
    }, [url]) 

    return state;


}
