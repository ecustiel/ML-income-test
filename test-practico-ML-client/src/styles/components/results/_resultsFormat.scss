


.resultsFormat__container{
    padding: 16px;
    display: flex;
    width: 72vw;
    background-color: white;
    border-bottom: 0.3px solid gray ;
}


.resultsFormat__imgContainer {
    display: inline;  
    text-align: left;
    margin-right: 16px;

    &-img{
        width: 180px;
        height: 180px;
    }
    
}

//container de la informacion
.resultsFormat {
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    width: 50%;
}

//precio e icono shipping
.resultsFormat_detsContainer-price{
    float: left;
    font-size: 24px;
    margin: 32px 10px 32px 16px;
}

.resultsFormat_detsContainer-img {
    float: left;
    margin-top: 35px;
    
}

//titulo
.resultsFormat__titleContainer {
    flex-wrap: wrap;
   display: flex;
   height: 50%;
   text-align: left;

   &-title {
    margin-left: 0;
    width: 100%;
    flex-wrap: wrap;
    font-size: 18px;
   }
    
}


//localidad
.resultsFormat__locContainer{
    display: flex;
    justify-content: center;
    width: 100%;
    margin-top: 5rem;
    
    &-loc {
        display: flex;
    justify-content: center;
    margin-right: 0;
    font-size: 12px;
    }
}


