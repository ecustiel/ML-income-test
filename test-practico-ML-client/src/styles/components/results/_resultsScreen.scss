

.resultsScreen__main {
    width: 100vw;
    align-items: center;
    text-align: center;
    
}

.resultsScreen__box {
    display: flex;
    width: 74vw;
    margin: 50px auto;
    height: 100h;
    background-color: $background-gray;
    border-color: black;
}