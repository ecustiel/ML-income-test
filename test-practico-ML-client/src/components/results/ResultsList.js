import React, {useState, useEffect} from 'react'
import {ResultsFormat} from './ResultsFormat';


export const ResultsList = ({items}) => {

    const [itemsArray, setitemsArray] = useState([])
    
    useEffect(() => {
        setitemsArray(items);
    }, [items]);
    
    return (
        <div>
            {
                itemsArray.map((item, index) =>  {
                    
                   return <ResultsFormat key={index}
                        {...item}
                        />
                }) 

                    }

        </div>
    )
}
