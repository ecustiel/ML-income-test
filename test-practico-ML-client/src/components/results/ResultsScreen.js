import React from 'react'
import {useLocation} from 'react-router-dom';
import { useFetch } from '../../hooks/useFetch';
import {ResultsList} from './ResultsList';


export const ResultsScreen = () => {

    const location = useLocation();
    const query = location.search.split('=');
    const queryToSearch = query[1];
    
    const {data, loading, error} =  useFetch(`/api/items?q=:${encodeURI(queryToSearch)}`);
    console.log(data);
    
    
    
    
    const itemToSend = [];

    if(loading === false){
     for (let i = 0; i < data.results.length; i++) {
        
        if (i < 4) {
           itemToSend.push({
               name: 'Eduardo',
               lastname: 'Custiel',
               categories:[data.results[i].category_id],
               items: [{
                 id: data.results[i].id,
                 title: data.results[i].title,
                 price: {
                     currency: data.results[i].currency_id,
                     amount: data.results[i].price
                 },
                 picture: data.results[i].thumbnail,
                 condition:  data.results[i].condition,
                 free_shipping: data.results[i].shipping.free_shipping,
                 loc: data.results[i].address.state_name
               }]
           });
        }
    
    }   
    }

    return (
        <div className='resultsScreen__main'>
            <div className='resultsScreen_path'>
                <span className='resultsScreen_path-text'></span>
            </div>
            <div className='resultsScreen__box'>
                
                        <ResultsList items={itemToSend} />
                      
            </div>
            
        </div>
    )
}
