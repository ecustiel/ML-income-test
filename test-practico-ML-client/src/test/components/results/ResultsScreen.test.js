import React from 'react';
import { shallow } from 'enzyme';
import {renderHook} from '@testing-library/react-hooks';
import {ResultsScreen} from '../../../components/results/ResultsScreen';
import { useFetch } from '../../../hooks/useFetch';
jest.mock('../../../components/results/ResultsScreen')


describe('Haciendo pruebas del componente <ResultsScreen />', () => {
    
    const location = {hash: "",
    key: "97q50b",
    pathname: "/items",
    search: "?search=ipad",
    state: undefined}
    
    

    test('Debe mostrar el componente <ResultsScreen />', () => {

      location.pathname='/item';
      const wrapper = shallow(<ResultsScreen />);
      expect(wrapper).toMatchSnapshot(); 
    })

    test('Usando el useFetch', async() => {
        
       const {result, waitForNextUpdate} = renderHook(() => useFetch('https://api.mercadolibre.com/sites/MLA/search?q=iphone'));
        await waitForNextUpdate();
        
       const {data, loading, error} = result.current;
       expect(data).not.toBe({});
       expect(loading).toBe(false);
       expect(error).toBe(null);
       
        
    })
    
    test('Debe fallar ya que la url es incorrecta', async() => {
        const {result, waitForNextUpdate} = renderHook(() => useFetch('https://api.mercadolibre.com/sites/MLAs/search?q=iphone'));
        await waitForNextUpdate();
        
        
        

       const {data, loading, error} = result.current;
       expect(data).not.toBe(null);
       expect(loading).toBe(false);
       expect(error).toBe('No se pudo cargar la informacion');
    })
    
    
})
