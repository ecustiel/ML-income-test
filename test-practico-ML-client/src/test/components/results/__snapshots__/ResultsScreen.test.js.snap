// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Haciendo pruebas del componente <ResultsScreen /> Debe mostrar el componente <ResultsScreen /> 1`] = `""`;
